packages/records/lib/models/records.integration.test.ts

```diff
@@ -431,134 +466,120 @@ describe('Records service', () => {
             });
+            expect(next_cursor).toBe(null); // no next page
+        });
 
-            if (response.isErr() || !response.value) {
-                throw new Error('Fail to fetch records');
-            }
+        it('Should paginate the records to retrieve all records', async () => {
+            const numOfRecords = 3000;
+            const limit = 100;
+            const { connectionId, model } = await upsertNRecords(numOfRecords);
 
-            const { records, next_cursor } = response.value;
+            let cursor: string | undefined | null = null;
+            const allFetchedRecords = [];
+            do {
+                const response = await Records.getRecords({
+                    connectionId,
+                    model,
+                    limit,
+                    ...(cursor && { cursor })
+                });
 
-            allFetchedRecords.push(...records);
+                if (response.isErr() || !response.value) {
+                    throw new Error('Fail to fetch records');
+                }
 
-            cursor = next_cursor;
+                const { records, next_cursor } = response.value;
 
-            expect(records).not.toBe(undefined);
-            expect(records?.length).toBeLessThanOrEqual(limit);
-        } while (cursor);
+                allFetchedRecords.push(...records);
 
-        for (let i = 1; i < allFetchedRecords.length; i++) {
-            const currentRecordDate = dayjs(allFetchedRecords[i]?._nango_metadata.first_seen_at);
-            const previousRecordDate = dayjs(allFetchedRecords[i - 1]?._nango_metadata.first_seen_at);
+                cursor = next_cursor;
 
-            expect(currentRecordDate.isAfter(previousRecordDate) || currentRecordDate.isSame(previousRecordDate)).toBe(true);
-        }
-        expect(allFetchedRecords.length).toBe(numOfRecords);
-    });
+                expect(records).not.toBe(undefined);
+                expect(records?.length).toBeLessThanOrEqual(limit);
+            } while (cursor);
+
+            for (let i = 1; i < allFetchedRecords.length; i++) {
+                const currentRecordDate = dayjs(allFetchedRecords[i]?._nango_metadata.first_seen_at);
+                const previousRecordDate = dayjs(allFetchedRecords[i - 1]?._nango_metadata.first_seen_at);
 
-    it('Should be able to retrieve 20K records in under 5s with a cursor', async () => {
-        const numOfRecords = 20000;
-        const limit = 1000;
-        const { connectionId, model } = await upsertNRecords(numOfRecords);
+                expect(currentRecordDate.isAfter(previousRecordDate) || currentRecordDate.isSame(previousRecordDate)).toBe(true);
+            }
+            expect(allFetchedRecords.length).toBe(numOfRecords);
+        });
 
-        let cursor: string | undefined | null = null;
-        let allRecordsLength = 0;
+        it('Should retreive records by external_id', async () => {
```

Comment:
Sorry, git really went insane on this diff. The blue below is the actual new test, but it may be easier to look at it side by side or in an editor.